x-pack/plugins/upgrade_assistant/server/lib/telemetry/usage_collector.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { set } from 'lodash';
+import { SavedObjectsRepository } from 'src/server/saved_objects/service/lib/repository';
+import {
+  UPGRADE_ASSISTANT_DOC_ID,
+  UPGRADE_ASSISTANT_TYPE,
+  UpgradeAssistantTelemetry,
+  UpgradeAssistantTelemetrySavedObject,
+  UpgradeAssistantTelemetrySavedObjectAttributes,
+  UpgradeAssistantTelemetryServer,
+} from '../../../common/types';
+import { isDeprecationLoggingEnabled } from '../es_deprecation_logging_apis';
+
+async function getSavedObjectAttributesFromRepo(
+  savedObjectsRepository: SavedObjectsRepository,
+  docType: string,
+  docID: string
+) {
+  try {
+    return (await savedObjectsRepository.get(docType, docID)).attributes;
+  } catch (e) {
+    return null;
+  }
+}
+
+export async function fetchUpgradeAssistantMetrics(
+  callCluster: any,
+  server: UpgradeAssistantTelemetryServer
+): Promise<UpgradeAssistantTelemetry> {
+  const { getSavedObjectsRepository } = server.savedObjects;
+  const savedObjectsRepository = getSavedObjectsRepository(callCluster);
+  const upgradeAssistantSOAttributes = await getSavedObjectAttributesFromRepo(
+    savedObjectsRepository,
+    UPGRADE_ASSISTANT_TYPE,
+    UPGRADE_ASSISTANT_DOC_ID
+  );
+  const loggerDeprecationCallResult = await callCluster('cluster.getSettings', { ignore: [404] });
```

Comment:
I'm not sure that calling `GET /_cluster/settings` would ever return a 404, or if it would that we would want to ignore that.

Also, should this have the include_defaults parameter? https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-get-settings.html#cluster-get-settings

Reply:
@tsullivan thanks for the review. I just included the `include_defaults` parameter. Regarding the 404 return I think you are right. I just removed the ignore and wrapped the callCluster into a function!